astroquery/heasarc/core.py

```diff
@@ -1,92 +1,247 @@
-# Licensed under a 3-clause BSD style license - see LICENSE.rst
 
-from typing import Union
+import os
+
+import shutil
+import requests
+import tarfile
 import warnings
-from io import StringIO, BytesIO
+import numpy as np
 from astropy.table import Table
-from astropy.io import fits
 from astropy import coordinates
 from astropy import units as u
-from ..query import BaseQuery
-from ..utils import commons
-from ..utils import async_to_sync
+from astropy.utils.decorators import deprecated, deprecated_renamed_argument
+
+import pyvo
+
+from astroquery import log
+from ..query import BaseQuery, BaseVOQuery
+from ..utils import commons, async_to_sync, parse_coordinates
 from ..exceptions import InvalidQueryError, NoResultsWarning
 from . import conf
 
-__all__ = ['Heasarc', 'HeasarcClass']
 
+@async_to_sync
```

Comment:
As this is practically a new module, no need to do this. (I'll clean these up in the upcoming astroquery work, but it's even better if it's not added).

We have started to talk about to ditch the duplication of methods and implement actual async methods (https://github.com/astropy/astroquery/issues/2598)

Reply:
removed!